sdk/storage/src/datalake/service_client.cpp

```diff
@@ -0,0 +1,165 @@
+  }
+
+  ServiceClient::ServiceClient(
+    const std::string& serviceUri,
+    std::shared_ptr<SharedKeyCredential> credential,
+    const ServiceClientOptions& options)
+      : m_serviceUri(serviceUri)
+  {
+    std::vector<std::unique_ptr<Azure::Core::Http::HttpPolicy>> policies;
+    for (const auto& p : options.policies)
+    {
+      policies.emplace_back(std::unique_ptr<Azure::Core::Http::HttpPolicy>(p->Clone()));
+    }
+    policies.emplace_back(std::make_unique<CommonHeadersRequestPolicy>());
+    policies.emplace_back(std::make_unique<SharedKeyPolicy>(credential));
+    policies.emplace_back(std::make_unique<Azure::Core::Http::TransportPolicy>(
+        std::make_shared<Azure::Core::Http::CurlTransport>()));
+    m_pipeline = std::make_shared<Azure::Core::Http::HttpPipeline>(policies);
+  }
+
+  ServiceClient::ServiceClient(
+    const std::string& serviceUri,
+    std::shared_ptr<TokenCredential> credential,
+    const ServiceClientOptions& options)
+      : m_serviceUri(serviceUri)
+  {
+    std::vector<std::unique_ptr<Azure::Core::Http::HttpPolicy>> policies;
+    for (const auto& p : options.policies)
+    {
+      policies.emplace_back(std::unique_ptr<Azure::Core::Http::HttpPolicy>(p->Clone()));
+    }
+    policies.emplace_back(std::make_unique<CommonHeadersRequestPolicy>());
+    // not implemented yet
+    unused(credential);
+    policies.emplace_back(std::make_unique<Azure::Core::Http::TransportPolicy>(
+        std::make_shared<Azure::Core::Http::CurlTransport>()));
+    m_pipeline = std::make_shared<Azure::Core::Http::HttpPipeline>(policies);
+  }
+
+  ServiceClient::ServiceClient(
+      const std::string& serviceUri,
+      const ServiceClientOptions& options)
+      : m_serviceUri(serviceUri)
+  {
+    std::vector<std::unique_ptr<Azure::Core::Http::HttpPolicy>> policies;
+    for (const auto& p : options.policies)
+    {
+      policies.emplace_back(std::unique_ptr<Azure::Core::Http::HttpPolicy>(p->Clone()));
+    }
+    policies.emplace_back(std::make_unique<CommonHeadersRequestPolicy>());
+    policies.emplace_back(std::make_unique<Azure::Core::Http::TransportPolicy>(
+        std::make_shared<Azure::Core::Http::CurlTransport>()));
+    m_pipeline = std::make_shared<Azure::Core::Http::HttpPipeline>(policies);
+  }
+
+  FileSystemClient ServiceClient::GetFileSystemClient(
+      const std::string& fileSystemName) const
+  {
+    FileSystemClient client = FileSystemClient();
+    auto builder = m_serviceUri; // Copy
+    builder.AppendPath(fileSystemName, true);
+    client.m_fileSystemUri = std::move(builder);
+    client.m_pipeline = m_pipeline; // Copy
+    return client;
+  }
+
+  ServiceListFileSystemsResponse ServiceClient::ListFileSystems(
+    const ListFileSystemsOptions& options) const
+  {
+    DataLakeRestClient::Service::ListFileSystemsOptions protocolLayerOptions;
+    // TODO: Add null check here when Nullable<T> is supported
+    // TODO: Add ApiVersion injection.
```

Comment:
How to inject?

Reply:
Removed, too bad we don't have this per API level anymore.